src/Veldrid/OpenGL/OpenGLPipeline.cs

```diff
@@ -239,6 +259,27 @@ private void ProcessResourceSetLayouts(ResourceLayout[] layouts)
                         resourceNamePtr[byteCount - 1] = 0; // Add null terminator.
                         int location = glGetUniformLocation(_program, resourceNamePtr);
                         CheckLastError();
+#if DEBUG
+                        if (location == -1)
+                        {
+                            uint uniformIndex = 0;
+                            var names = new List<string>();
+                            do
+                            {
+                                uint bufferNameByteCount = 64;
+                                byte* bufferNamePtr = stackalloc byte[(int) bufferNameByteCount];
+                                uint actualLength;
+                                int size;
+                                uint type;
+                                glGetActiveUniform(_program, uniformIndex, bufferNameByteCount,
+                                    &actualLength, &size, &type, bufferNamePtr);
+                                string name = Encoding.UTF8.GetString(bufferNamePtr, (int) actualLength);
```

Comment:
glGetError is already being called in the do/while condition at the end of the block, is there any issue with doing it that way?

Reply:
Checking at the end of each do/while iteration is too late, because you've already constructed the string and placed it into the `names` list, to be used later. If an OpenGL error occurred, then there is likely no valid string data to use. I actually caught this because when I ran this code under the debugger, there was always an extra garbage-filled string at the end of the list.

Reply:
Ah, of course… if it's failed then we can't trust anything it returned. That also explains why I was usually seeing the last name duplicated when testing, I should've investigated further rather than just throwing a .Distinct() on it. I've submitted a fixed version now.